MergerService/Utils/TaskUtils.cs

```diff
@@ -13,107 +14,154 @@ public class TaskUtils : ITaskUtils
         private IConfigurationManager _configuration;
         private ILogger _logger;
         private ActivitySource _activitySource;
-
         private int _maxAttempts;
+        private JsonSerializerSettings _jsonSerializerSettings;
+        private string _overseerUrl;
+        private string _jobManagerUrl;
 
-        public TaskUtils(int maxAttempts, IConfigurationManager configuration, IHttpRequestUtils httpClient, ILogger<TaskUtils> logger, ActivitySource activitySource)
+        public TaskUtils(IConfigurationManager configuration, IHttpRequestUtils httpClient, ILogger<TaskUtils> logger, ActivitySource activitySource)
         {
             this._httpClient = httpClient;
             this._configuration = configuration;
             this._logger = logger;
             this._activitySource = activitySource;
-            this._maxAttempts = maxAttempts;
+            this._maxAttempts = this._configuration.GetConfiguration<int>("TASK", "maxAttempts");
+
+            // Construct Json serializer settings
+            _jsonSerializerSettings = new JsonSerializerSettings();
+            _jsonSerializerSettings.Converters.Add(new StringEnumConverter());
+
+            _overseerUrl = this._configuration.GetConfiguration("TASK", "overseerUrl");
+            _jobManagerUrl = this._configuration.GetConfiguration("TASK", "jobManagerUrl");
             //TODO: add tracing
         }
 
         public MergeTask? GetTask(string jobType, string taskType)
         {
-            string baseUrl = this._configuration.GetConfiguration("TASK", "jobManagerUrl");
-            string relativeUri = $"tasks/{jobType}/{taskType}/startPending";
-            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
-            string taskData = this._httpClient.PostDataString(url, null, false);
+            // TODO: add heartbeat start method
 
-            if (taskData is null)
+            using (var dequeueActivity = this._activitySource.StartActivity("dequeue task"))
```

Comment:
save activity into variable only if  you there metadata should be add to the trace,
otherwise just use:
using (this._activitySource.StartActivity("dequeue task"))

Reply:
Removed.